Orange-OpenSource/Orange-Boosted-Bootstrap
# Release Boosted 5.1.0

## Tasks

- [x] [Bootstrap proposal PR](https://github.com/twbs/bootstrap/pull/34769) has been merged so I can keep my modification in this PR. One less thing to merge the next time :)
- [x] Designers strongly advise to remove [Add new offcanvas support to navbars](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/772/commits/fbe1c0be35ec3ef7f2c974225f507742bf1f4e64) from the documentation.
- [x] Remove in all files `aria-label="Close"` and replace it by `<span class="visually-hidden">Close</span>` inside the buttons
- ~~At the end, regenerate the `package-lock.json`~~

> Yeah, I just recreate it:
>
> git clean -dfx
> rm package-lock.json or del package-lock.json
> npm i
>
> Just be careful because in some cases the package ranges in package.json will install a newer version (at the time of writing autoprefixer for example). In those cases, just temporarily pin the dep.
>
> This is needed unfortunately because when using dependabot over time the package-lock.json file becomes bloated. I haven't switched to npm 7.x to see if the issue is fixed there.

- [x] Fix d4664f4913843f9f099ed7081fcedcd58736fe87 from [e72916e5b741b07b94b12b3707129e868811bdbb](https://github.com/twbs/bootstrap/commit/e72916e5b741b07b94b12b3707129e868811bdbb): work to do on text/background opacities
- [x] Simplify code from https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/772#discussion_r699001080
- [x] Fix bundlewatch

## Release

  - Firefox ESR
  - IE11 (v4 only)
  - Latest Edge, Chrome, Firefox, Safari
  - iOS Safari
  - Chrome & Firefox on Android
  - then, if bumping a minor or major version:
    - [ ] Ensure Algolia indexes new release content ([probably requires a PR](https://github.com/algolia/docsearch-configs/pull/1954))
  - check wrong matches in `CHANGELOG.md`, and maybe `site/content/docs/<version>/migration.md`
  - :warning: check the `package-lock.json` content, only "boosted" should have its version changed!
  - install [Conventionnal Changelog](https://github.com/conventional-changelog/conventional-changelog) and `conventional-changelog-cli` globally
  - run `conventional-changelog -p angular -i CHANGELOG.md -s`
  - and probably maintain [a ship list (eg for v4.4.0)](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/226)
- [ ] merge (on `v4-dev` or `v5-dev`)
  - attach zip file
  - paste CHANGELOG / Ship list in the release's description
- [ ] run `build/ship.sh x.x.x` to… (v4 only)
  - Pull the latest changes update new version numbers
  - Compile CSS and JS
  - Compress dist files
  - Generate the hashes and update our config's cdn section
  - Compile the docs
  - Copy-pasta built docs to `bs-docs`-named clone of Boosted.
  - if you are already logged in NPM (with a personnal account, for example), [you'd better use a repository scoped `.npmrc` file](https://stackoverflow.com/questions/30114166/how-to-have-multiple-npm-users-set-up-locally)
  - if you're releasing a pre-release, use `--tag`, eg for v5-alpha1 `npm publish boosted-5.0.0-alpha1.tgz --tag next`
  - [ ] copy `../_site` (v5-dev) or `../bs-docs` (v4-dev) to the `gh-pages` branch
- [ ] Designers asked for the [new "Placeholders" component](39b7c75b13dd982e1a4f106f412ff18acc0026ec) to try to improve the glow animation with the following steps #eee -> #ddd -> #ccc -> #999 if possible, and to try to improve the wave animation which is not visible in this first version. Placeholders use opacity so this proposal needs to be studied more in depth --> Create an issue in GitHub to tackle this subject and improve the rendering of the placeholders
